classes/mail/MailTemplate.inc.php

```diff
@@ -146,30 +146,30 @@ function assignParams($params = array()) {
 		if ($this->context) {
 			// Add context-specific variables
 			$dispatcher = $application->getDispatcher();
-			$params = array_merge(array(
-				'principalContactSignature' => $this->context->getData('contactName'),
-				'contextName' => $this->context->getLocalizedName(),
+			$params = array_merge([
+				'principalContactSignature' => htmlspecialchars($this->context->getData('contactName')),
+				'contextName' => htmlspecialchars($this->context->getLocalizedName()),
 				'contextUrl' => $dispatcher->url($request, ROUTE_PAGE, $this->context->getPath()),
-			), $params);
+			], $params);
 		} else {
 			// No context available
-			$params = array_merge(array(
-				'principalContactSignature' => $site->getLocalizedContactName(),
-			), $params);
+			$params = array_merge([
+				'principalContactSignature' => htmlspecialchars($site->getLocalizedContactName()),
+			], $params);
 		}
 
 		if (!defined('SESSION_DISABLE_INIT') && ($user = $request->getUser())) {
 			// Add user-specific variables
-			$params = array_merge(array(
-				'senderEmail' => $user->getEmail(),
-				'senderName' => $user->getFullName(),
-			), $params);
+			$params = array_merge([
+				'senderEmail' => htmlspecialchars($user->getEmail()),
```

Comment:
Although unlikely in practice, this could potentially cause problems with some email addresses, because the email address spec is really loose. I'd be happy limiting it anyway, but I wonder if we should try to do it before the address goes into the database. Maybe something to tighten up for 3.4.

Reply:
(same as: https://github.com/pkp/pkp-lib/pull/7858#discussion_r856476741)